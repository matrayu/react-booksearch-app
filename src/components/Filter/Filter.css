.filter {
    padding: 5px;
}

.form label {
    padding: 10px;
}