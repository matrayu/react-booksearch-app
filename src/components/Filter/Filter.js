import React, { Component } from 'react';
import './Filter.css';

class Filter extends Component {
  handlePrintType = (e) => {
    this.filterPrintType(e.target.value)
  };

  filterPrintType = (printType) => {
    this.props.handlePrintType(printType)
  };

  handleBookType = (e) => {
    this.filterBookType(e.target.value)
  };

  filterBookType = (bookType) => {
    this.props.handleBookType(bookType)
  };

  render() {
    return (
      <div className="filter">
        <form className="filter__form">
            <label>Print Type:
                <select defaultValue={this.props.printType} onChange={this.handlePrintType}>
                    <option value="all">All</option>
                    <option value="books">Books</option>
                    <option value="magazines">Magazines</option>
                </select>    
            </label> 
            <label>Book Type: 
                <select defaultValue={this.props.bookType} onChange={this.handleBookType}>
                    <option value="no-filter">No Filter</option>
                    <option value="ebooks">eBooks</option>
                    <option value="free-ebooks">Free eBooks</option>
                    <option value="full">Full</option>
                    <option value="paid-ebooks">Paid eBooks</option>
                    <option value="partial">Partial</option>
                </select>    
            </label> 
        </form>
      </div>
    );
  }
}

export default Filter;