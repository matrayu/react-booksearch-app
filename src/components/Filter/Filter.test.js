import React from 'react';
import ReactDOM from 'react-dom';
import Filter from './FIlter';
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Filter />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
      .create(<Filter />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});