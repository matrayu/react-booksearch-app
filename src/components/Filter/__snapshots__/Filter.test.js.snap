// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="filter"
>
  <form
    className="filter__form"
  >
    <label>
      Print Type:
      <select
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="books"
        >
          Books
        </option>
        <option
          value="magazines"
        >
          Magazines
        </option>
      </select>
    </label>
    <label>
      Book Type:
      <select
        onChange={[Function]}
      >
        <option
          value="no-filter"
        >
          No Filter
        </option>
        <option
          value="ebooks"
        >
          eBooks
        </option>
        <option
          value="free-ebooks"
        >
          Free eBooks
        </option>
        <option
          value="full"
        >
          Full
        </option>
        <option
          value="paid-ebooks"
        >
          Paid eBooks
        </option>
        <option
          value="partial"
        >
          Partial
        </option>
      </select>
    </label>
  </form>
</div>
`;
